backend/src/core/middlewares/unsubscriber.middleware.ts

```diff
@@ -38,67 +42,30 @@ const isUnsubscribeRequestValid = async (
   }
 }
 
-/**
- * Retrieves the unsubscribe status and the associated campaign
- * @param _req
- * @param res
- */
-const getUnsubscriberStatus = async (
-  _req: Request,
-  res: Response
-): Promise<Response> => {
-  const { campaign, recipient } = res.locals.unsubscribe
-  const unsubscribe = await UnsubscriberService.getUnsubscriber(
-    campaign.id,
-    recipient
-  )
-
-  return res.json({
-    unsubscribed: unsubscribe !== null,
-    campaign: {
-      name: campaign.name,
-      channelType: campaign.type,
-    },
-  })
-}
-
 /**
  * Creates a new unsubscribe record
  * @param _req
  * @param res
  * @param next
  */
-const createUnsubscriber = async (
+const findOrCreateUnsubscriber = async (
   _req: Request,
-  res: Response,
-  next: NextFunction
+  res: Response
 ): Promise<Response | void> => {
-  try {
-    const { campaign, recipient } = res.locals.unsubscribe
-    await UnsubscriberService.createUnsubscriber({
-      campaignId: campaign.id,
-      recipient,
-    })
-    return res.status(201).json({
-      unsubscribed: true,
-      campaign: {
-        name: campaign.name,
-        channelType: campaign.type,
-      },
-    })
-  } catch (err) {
-    if (err instanceof UniqueConstraintError) {
-      return res.status(400).json({
-        message: 'Campaign is already unsubscribed',
-      })
-    }
+  const { campaignId, recipient } = res.locals.unsubscriber
+  const { 1: created } = await UnsubscriberService.findOrCreateUnsubscriber({
+    campaignId,
+    recipient,
+  })
 
-    next(err)
-  }
+  const statusCode = created ? 201 : 200
+  return res.status(statusCode).json({
```

Comment:
Actually according to Sarah's design we don't need campaignId or recipient at all, right? Sorry should have mentioned it along with `created_at`

Reply:
No worries, you're right that it's redundant. I've remove the response body before merging. Thanks!